File: app/src/main/java/com/future/awaker/main/MainFragment.java
package com.future.awaker.main;

import android.databinding.BindingAdapter;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;
import com.future.awaker.R;
import com.future.awaker.base.BaseListFragment;
import com.future.awaker.base.listener.OnItemClickListener;
import com.future.awaker.data.New;
import com.future.awaker.databinding.FragMainBinding;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by ruzhan on 2017/7/6.
 */

public class MainFragment extends BaseListFragment<FragMainBinding> implements OnItemClickListener<New> {

    private MainViewModel mainViewModel;
    private MainAdapter adapter;

    public static MainFragment newInstance() {
        return new MainFragment();
    }

    @Override
    protected int getLayout() {
        return R.layout.frag_main;
    }

    @Override
    protected int getEmptyLayout() {
        return R.layout.layout_empty;
    }

    @Override
    protected void initData() {
        setBaseListViewModel(mainViewModel);
        binding.setViewmodel(mainViewModel);

        adapter = new MainAdapter(this);
        binding.recyclerView.setAdapter(adapter);
        mainViewModel.setToken("f32b30c2a289bfca2c9857ffc5871ac8", 0);
    }

    @Override
    public void onDestroyView() {

        super.onDestroyView();
    }


    @BindingAdapter({"news"})
    public static void setNews(RecyclerView recyclerView, List<New> news) {
        RecyclerView.Adapter adapter = recyclerView.getAdapter();
        if (adapter instanceof MainAdapter) {
            ((MainAdapter)adapter).setData(new ArrayList<>(news));
        }
    }

    public void setMainViewModel(MainViewModel mainViewModel) {
        this.mainViewModel = mainViewModel;
    }

    @Override
    public void onItemClick(View view, int position, New bean) {
        Toast.makeText(getContext(), position + " : " + bean.title, Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/future/awaker/MainFragment.kt
package com.future.awaker

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.future.day.DayHomeFragment
import com.ruzhan.movie.home.MovieHomeFragment
import kotlinx.android.synthetic.main.frag_main.*

class MainFragment : Fragment() {

    companion object {

        @JvmStatic
        fun newInstance() = MainFragment()
    }

    private val fragmentMap = HashMap<String, Fragment>()

    private var dayHomeFragment: DayHomeFragment? = null
    private var movieHomeFragment: MovieHomeFragment? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.frag_main, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        bottomNavigationView.setOnNavigationItemSelectedListener {
            if (bottomNavigationView.selectedItemId != it.itemId) {
                replaceFragment(it.itemId)
            }
            true
        }
        replaceFragment(R.id.article)
    }

    private fun replaceFragment(tabId: Int) {
        val fm = childFragmentManager
        val transaction = fm.beginTransaction()
        for (frag in fragmentMap.values) {
            transaction.hide(frag)
        }
        var fragTag: String? = null
        var frag: Fragment? = null
        when (tabId) {
            R.id.article -> {
                fragTag = "DayHomeFragment"
                frag = fragmentMap[fragTag]

                if (frag == null) {
                    frag = DayHomeFragment.newInstance()
                    dayHomeFragment = frag
                    transaction.add(R.id.container, frag, fragTag)

                } else {
                    transaction.show(frag)
                }

            }
            R.id.movie -> {
                fragTag = "MovieHomeFragment"
                frag = fragmentMap[fragTag]

                if (frag == null) {
                    frag = MovieHomeFragment.newInstance()
                    movieHomeFragment = frag
                    transaction.add(R.id.container, frag, fragTag)

                } else {
                    transaction.show(frag)
                }
            }
        }
        if (fragTag != null && frag != null) {
            fragmentMap[fragTag] = frag
        }
        if (!fm.isDestroyed) {
            transaction.commitAllowingStateLoss()
        }
    }
}